util/sock.c

```diff
@@ -266,12 +267,8 @@ sock_resolve(const char * addr)
 	ips = &s[1];
 	ips[strlen(ips) - 1] = '\0';
 
-	/*
-	 * Parse the port number.  If strtol fails to parse the port number,
-	 * it will return 0; but that's fine since port 0 is invalid anyway.
-	 */
-	p = strtol(ports, NULL, 10);
-	if ((p <= 0) || (p >= 65536)) {
+	/* Parse the port number, allowing for trailing characters. */
+	if (PARSENUM_EX(&p, ports, 1, 65535, 10, 1)) {
```

Comment:
Since we've got this code now, we might as well use it here.  I tested it in spiped (since libcperciva doesn't have any test code that calls `sock_resolve`).

Reply:
Hmm, I don't think we need to support trailing junk.  Are you aware of any case where that support is needed?

Reply:
You're right, we don't need it here.  I was thinking about parsing chunked HTTP headers.